src/hotspot/share/interpreter/zero/bytecodeInterpreter.cpp

```diff
@@ -1995,13 +1995,12 @@ void BytecodeInterpreter::run(interpreterState istate) {
               //   - in debug mode, ThreadLocalAllocBuffer::allocate mangles
               //     this area, and we still need to initialize it
               if (DEBUG_ONLY(true ||) !ZeroTLAB) {
-                size_t hdr_size = oopDesc::header_size();
-                Copy::fill_to_words(result + hdr_size, obj_size - hdr_size, 0);
+                size_t hdr_size = instanceOopDesc::base_offset_in_bytes();
```

Comment:
For Zero case, I think we can just `Copy::fill_to_words` the entire object: the overhead of dealing with header math is likely comparable to the overhead of additional header writes and the benefits the word-sized clearing brings. This can probably be separated in another PR that deals with zapping better.